gemma-core/src/main/resources/ubic/gemma/core/loader/expression/singleCell/transform/filter-10x-mex.py

```diff
@@ -0,0 +1,194 @@
+    features_tsv = ensure_str(_find_file_with_prefix(genome_dir, FEATURES_TSV_GZ, prefix))
+    matrix_mtx = ensure_str(_find_file_with_prefix(genome_dir, "matrix.mtx.gz", prefix))
+    
+    for filepath in [barcodes_tsv, features_tsv, matrix_mtx]:
+        if not os.path.exists(filepath):
+            raise OSError(f"Required file not found: {filepath}")
+    barcodes = pd.read_csv(
+        barcodes_tsv, delimiter="\t", header=None, usecols=[0], dtype=bytes
+    ).values.squeeze()
+    features = pd.read_csv(features_tsv, delimiter="\t", header=None)
+
+    feature_defs = []
+    tags = {"genome": genome_name}
+    for idx, (_, r) in enumerate(features.iterrows()):
+        fd = FeatureDef(idx, r[0], r[1], r[2], tags)
+        feature_defs.append(fd)
+
+    feature_ref = FeatureReference(feature_defs, [])
+
+    matrix = sp_io.mmread(matrix_mtx)
+    # make csc
+    matrix = matrix.tocsc()
+    mat = CountMatrix(feature_ref, barcodes, matrix)
+    return mat
+
+def detect_gem_groups(mtx):
+    gem_groups = set()
+    for bc in mtx.bcs:
+        gem_group = get_gem_group_from_barcode(bc)
+        gem_groups.add(gem_group)
+        
+    gem_groups = sorted(gem_groups)
+    return gem_groups
+
+
+def save_features_tsv(feature_ref, base_dir, compress):
+    """Save a FeatureReference to a tsv file."""
+    out_features_fn = os.path.join(ensure_binary(base_dir), b"features.tsv")
+    if compress:
+        out_features_fn += b".gz"
+
+    with cr_io.open_maybe_gzip(out_features_fn, "wb") as f:
+        for feature_def in feature_ref.feature_defs:
+            f.write(feature_def.id if isinstance(feature_def.id, bytes) else feature_def.id.encode())
+            f.write(b"\t")
+            f.write(ensure_binary(feature_def.name))
+            f.write(b"\t")
+            f.write(ensure_binary(feature_def.feature_type))
+            f.write(b"\n")
+
+
+def main():
+    mtx_dir = sys.argv[1] 
+    outdir = sys.argv[2] 
+    genome_name = sys.argv[3]
+    # make chem optional
+    if len(sys.argv) > 4:
+        chem = sys.argv[4]
+    else:
+        chem = None
+    #
+    if not mtx_dir and not outdir and not genome_name:
+        raise ValueError("Both mtx_dir, outdir and genome_name must be provided.")
+
+    prefix = detect_prefix(mtx_dir, "barcodes.tsv.gz")
+    if prefix is None:
+        print(f"Detected prefix: {prefix}")
+    mtx = load_mtx(mtx_dir=mtx_dir, genome_name=genome_name, prefix=prefix)
+
+    gem_groups = detect_gem_groups(mtx)
+    print(f"Detected gem groups: {gem_groups}")
+    sample = prefix if prefix else os.path.basename(mtx_dir)
+
+    filtered_metrics_groups, filtered_bcs_groups = call_initial_cells(
+        matrix=mtx,
+        genomes=[genome_name],
+        sample=sample,
+        unique_gem_groups=gem_groups,
+        method=FilterMethod.ORDMAG_NONAMBIENT, 
+        recovered_cells=None,
+        cell_barcodes=None,
+        force_cells=None,
```

Comment:
@rachadele according to the type annotations, `cell_bracodes` and `force_cells` must be set. Is there anything we can do about it?